src/Dhall/Import.hs

```diff
@@ -225,23 +226,25 @@ instance Show ReferentiallyOpaque where
 
 -- | Extend another exception with the current import stack
 data Imported e = Imported
-    { importStack :: [Import] -- ^ Imports resolved so far, in reverse order
-    , nested      :: e        -- ^ The nested exception
+    { importStack :: NonEmpty Import -- ^ Imports resolved so far, in reverse order
+    , nested      :: e               -- ^ The nested exception
     } deriving (Typeable)
 
 instance Exception e => Exception (Imported e)
 
 instance Show e => Show (Imported e) where
     show (Imported imports e) =
-            (case imports of [] -> ""; _ -> "\n")
-        ++  unlines (map indent imports')
-        ++  show e
+           concat (zipWith indent [0..] toDisplay)
+        ++ show e
       where
-        indent (n, import_) =
-            take (2 * n) (repeat ' ') ++ "↳ " ++ Dhall.Pretty.Internal.prettyToString import_
-        -- Canonicalize all imports
-        imports' = zip [0..] (drop 1 (reverse (canonicalizeAll imports)))
-
+        indent n import_ =
+            "\n" ++ replicate (2 * n) ' ' ++ "↳ " ++ Dhall.Pretty.Internal.prettyToString import_
+        canonical = NonEmpty.toList (canonicalizeAll imports)
+        -- The immediate error's source is displayed by 'e', so we
+        -- need to not display the first (innermost) import, and the
+        -- final (outermost) import is fake to establish the base
+        -- directory. Also, we need outermost-first.
+        toDisplay = drop 1 (reverse (drop 1 canonical))
```

Comment:
The `doctest` error appears to be due to the new `drop 1` here.  According to the comment this appears to be intentional, but are you sure that the innermost import is already displayed by `e`?

Reply:
I thought it was! :( I'm having a look now too. That reasoning was based on the previous code having a `drop 1` in it but I might have confused myself while trying to get this function to work.

Reply:
Oh, it was on the wrong side of the `reverse` anyway for that. Now I don't know why the previous code had the `drop 1` in it at all.

Reply:
@quasicomputational: I think the reason it had the other `drop 1` is because it would implicitly insert `./.` as the root of the the chain of imports before your change and the `drop 1` removed it so it wouldn't show up in the chain of imports